### PR TITLE
Holocards and related functionality

### LINKED ISSUE #2172: Medical roles should be able to change the holocard of a marine in the health analyzer UI

### PR DESCRIPTION
**Pull request by blueDev2**
## About the PR

Added holocards, small icons that can be applied to people by medics doing triage
Fixes #2172

## Technical details

The Holocard status component is a networked component that tells the client which icon to show when a viewer is using a medical hud
The Holocard scanner component is a component attached to medical huds that allow medically trained individuals to examine a person and change the holocard on them
The health analyzer BUI has been changed to include a holocard change button and a holocard description. The holocard change button is disabed for non-medically trained individuals

## Media

https://github.com/CM-14/CM-14/assets/89804215/a6f3aae8-e7ad-4eb9-a0d1-8e5c58b85cca

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl: blueDev2
- add: Added Holocards

### CODE REVIEW

**Review by QuietlyWhisper — CHANGES_REQUESTED** [review]
Basic review for now.

Can you also add a space between the holocard change button and the description?

File: Resources/Prototypes/_CM14/Entities/Mobs/Species/base.yml [hunk 1]
```diff
@@ -105,6 +106,9 @@
         type: StrippableBoundUserInterface
       enum.CMSurgeryUIKey.Key:
         type: CMSurgeryBui
+      enum.HolocardChangeUIKey.Key:
+        type: HolocardChangeBoundUserInterface
+        interactionRange: 0
```
[the remark this answers is not in the document]
Comment (blueDev2) [reply Resources/Prototypes/_CM14/Entities/Mobs/Species/base.yml:111]: It's needed, else the health hud does not properly work when changing holocards. 
Now I could understand limiting interaction range to sight range but it really feels like splitting hairs here

File: Resources/Textures/_CM14/Interface/VerbIcons/ambulance.svg
[the remark this answers is not in the document]
Comment (blueDev2) [reply Resources/Textures/_CM14/Interface/VerbIcons/ambulance.svg:1]: I literally made it myself.
I looked at a specific icon when making it but it really does not look that similar.

File: Resources/Textures/_CM14/Interface/health_hud_holocard.rsi/meta.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "Taken from cmss13 at https://github.com/cmss13-devs/cmss13/blob/a68173bdb50716996547edcbcfea5b8cc18745b3/icons/mob/hud/hud.dmi",
```
[the remark this answers is not in the document]
Comment (blueDev2) [reply Resources/Textures/_CM14/Interface/health_hud_holocard.rsi/meta.json]: Yeah I just copied the meta file, I'll credit myself but um, should I just put it under my name or make a link to the applicable commit?

File: Resources/Prototypes/_CM14/Entities/Mobs/Species/base.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (QuietlyWhisper) [reply Resources/Prototypes/_CM14/Entities/Mobs/Species/base.yml:111]: I asked smug to look into it just in case but dont worry about this too much

**Comment by QuietlyWhisper** [comment]
> Yeah I just copied the meta file, I'll credit myself but um, should I just put it under my name or make a link to the applicable commit?

Yeah you can just use yourself as the source, link is not needed. Also applies for the two other images you made. You also get full credit for these images when someone uses our code to make a new server! That's why all assets need the copyrights text

**Comment by blueDev2** [comment]
image
Added spacing

**Comment by blueDev2** [comment]
> > Yeah I just copied the meta file, I'll credit myself but um, should I just put it under my name or make a link to the applicable commit?
> 
> Yeah you can just use yourself as the source, link is not needed. Also applies for the two other images you made. You also get full credit for these images when someone uses our code to make a new server, that's why all assets need the copyrights text (We also cannot use assets with no license)

Sorry, 2 other images? I think I only made 2 images, the ambulance and the holocard icons.

**Comment by blueDev2** [comment]
Also, is there any way for me to get the contributor role on the discord?

**Comment by QuietlyWhisper** [comment]
> Sorry, 2 other images? I think I only made 2 images, the ambulance and the holocard icons.

I mean those yes sorry for confusion.

> Also, is there any way for me to get the contributor role on the discord?
Ping in general, smug will review the ui code when she gets the time.

**Review by DrSmugleaf — APPROVED** [review]
Thanks for the PR!

**Pull request merged**